core/chain_makers.go

```diff
@@ -329,30 +330,25 @@ func GenerateVerkleChain(config *params.ChainConfig, parent *types.Block, engine
 			}
 
 			// Generate an associated verkle proof
-			if tr := statedb.GetTrie(); tr.IsVerkle() {
-				vtr := tr.(*trie.VerkleTrie)
-				// Generate the proof if we are using a verkle tree
-				// WORKAROUND: make sure all keys are resolved
-				// before building the proof. Ultimately, node
-				// resolution can be done with a prefetcher or
-				// from GetCommitmentsAlongPath.
-
-				keys := statedb.Witness().Keys()
-				for _, key := range keys {
-					out, err := vtr.TryGet(key)
-					if err != nil {
-						panic(err)
-					}
-					if len(out) == 0 {
-						panic(fmt.Sprintf("%x should be present in the tree", key))
-					}
-				}
-				vtr.Hash()
-				p, k, err := vtr.ProveAndSerialize(keys, statedb.Witness().KeyVals())
-				block.SetVerkleProof(p, k)
-				if err != nil {
-					panic(err)
-				}
+			tr := preState.GetTrie()
+			if !tr.IsVerkle() {
+				panic("tree should be verkle")
+			}
+
+			vtr := tr.(*trie.VerkleTrie)
+			// Make sure all keys are resolved before
+			// building the proof. Ultimately, node
+			// resolution can be done with a prefetcher
+			// or from GetCommitmentsAlongPath.
+			keys := statedb.Witness().Keys()
+			for _, key := range keys {
+				vtr.TryGet(key)
+			}
```

Comment:
What is the reason for doing this explicitly?  Are you trying to make sure the commitments are all computed (which appears to already be done from the call to `statedb.Commit()`)?

Reply:
it's necessary to make sure that all required nodes are loaded in memory - this is the _pre_ tree and we're starting off from  fresh tree